File: backend-csharp/Models/BienTheSanPham.cs
using System;
using System.Collections.Generic;

namespace WebFashion.Api.Models;

public partial class BienTheSanPham
{
    public long Id { get; set; }

    public long SanPhamId { get; set; }

    public string Sku { get; set; } = null!;

    public string? MauSac { get; set; }

    public string? DungLuong { get; set; }

    public string? Ram { get; set; }

    public decimal GiaGoc { get; set; }

    public decimal GiaBan { get; set; }

    public int TonKho { get; set; }

    public string? MaMauHex { get; set; }

    public string TrangThai { get; set; } = null!;

    public virtual ICollection<BaoHanh> BaoHanhs { get; set; } = new List<BaoHanh>();

    public virtual ICollection<BaoHiem> BaoHiems { get; set; } = new List<BaoHiem>();

    public virtual ICollection<ChiTietDonHang> ChiTietDonHangs { get; set; } = new List<ChiTietDonHang>();

    public virtual ICollection<ChiTietKiemKho> ChiTietKiemKhos { get; set; } = new List<ChiTietKiemKho>();

    public virtual ICollection<ChiTietPhieuNhap> ChiTietPhieuNhaps { get; set; } = new List<ChiTietPhieuNhap>();

    public virtual ICollection<GioHang> GioHangs { get; set; } = new List<GioHang>();

    public virtual ICollection<HinhAnhSanPham> HinhAnhSanPhams { get; set; } = new List<HinhAnhSanPham>();

    public virtual Kho? Kho { get; set; }

    public virtual SanPham SanPham { get; set; } = null!;
}

File: backend/models/BienTheSanPham.js
const { DataTypes } = require("sequelize");
const sequelize = require("../config/db");

const BienTheSanPham = sequelize.define(
  "BienTheSanPham",
  {
    id: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
    san_pham_id: { type: DataTypes.INTEGER, allowNull: false },
    sku: { type: DataTypes.STRING(50) },
    mau_sac: { type: DataTypes.STRING(50) },
    dung_luong: { type: DataTypes.STRING(50) },
    ram: { type: DataTypes.STRING(50) },
    gia_goc: { type: DataTypes.DECIMAL(18, 0) },
    gia_ban: { type: DataTypes.DECIMAL(18, 0) },
    ton_kho: { type: DataTypes.INTEGER, defaultValue: 0 },
    ma_mau_hex: { type: DataTypes.STRING(20) },
    trang_thai: { type: DataTypes.STRING(20), defaultValue: "active" },
  },
  {
    tableName: "BienTheSanPham",
    timestamps: false,
  },
);

module.exports = BienTheSanPham;
